frontend/src/modules/S3_Datasets/views/DatasetCreateForm.js

```diff
@@ -429,38 +429,45 @@ const DatasetCreateForm = (props) => {
                       <Card sx={{ mb: 3 }}>
                         <CardHeader title="Deployment" />
                         <CardContent>
-                          <TextField
-                            fullWidth
-                            error={Boolean(
-                              touched.environment && errors.environment
-                            )}
-                            helperText={
-                              touched.environment && errors.environment
-                            }
-                            label="Environment"
-                            name="environment"
-                            onChange={(event) => {
-                              setFieldValue('SamlGroupName', '');
-                              fetchGroups(
-                                event.target.value.environmentUri
-                              ).catch((e) =>
-                                dispatch({ type: SET_ERROR, error: e.message })
-                              );
-                              setFieldValue('environment', event.target.value);
+                          <Autocomplete
+                            id="environment"
+                            disablePortal
+                            options={environmentOptions.map((option) => option)}
+                            onChange={(event, value) => {
+                              setFieldValue('SamlAdminGroupName', '');
+                              setFieldValue('steward', '');
+                              if (value && value.environmentUri) {
+                                setFieldValue('environment', value);
+                                fetchGroups(value.environmentUri).catch((e) =>
+                                  dispatch({
+                                    type: SET_ERROR,
+                                    error: e.message
+                                  })
+                                );
+                              } else {
+                                setFieldValue('environment', '');
+                                setGroupOptions([]);
```

Comment:
Calling this out here but I think it will be the same to everywhere where we have `<Autocomplete>` of environment which is also setting `GroupOptions`

In my current testing when clicking `X` to delete `environment` value on form - the groupOptions delete but the `SamlAdminGroupName` value remains the same

May have to wrap `SamlAdminGroupName` in a similar way to how we do with `RequestAccessModal.js` as such:

```
                            <Box>
                              {groupOptions.length > 0 ? (
                                ... // CODE WE HAVE NOW
                              ) : (
                                <TextField
                                  error={Boolean(
                                    touched.groupUri && errors.groupUri
                                  )}
                                  helperText={
                                    touched.groupUri && errors.groupUri
                                  }
                                  fullWidth
                                  disabled
                                  label="Team"
                                  value="No teams found for this environment"
                                  variant="outlined"
                                />
                              )}
                            </Box>

Reply:
The PR comments are addressed in https://github.com/data-dot-all/dataall/pull/1347/commits/2298fa7f9a488edcd831b8da37be371e0e01f276. Although , I don't think the issue is completely fixed.

If instead of clicking in the `x` I select another environment the value shown in the form persists. The actual value passed to the API call and the value evaluated in the form is correct, it is just the value shown in the field.

In the next screenshot the group Scientist was selected for env1; then I selected the env wnv2 that does not have that group. I tried to submit the form, but the value for SamlAdminGroup is null. So the value is correct, but whatever is plotted int he form is not updated.

![image](https://github.com/data-dot-all/dataall/assets/71252798/9e9cc955-7adc-4cf6-8062-9ff15138be5b)

Reply:
It is fixed now with the commit: https://github.com/data-dot-all/dataall/pull/1347/commits/a094fbca8d643cf63ea1d18020e8630bbf07de15